oap-ape/ape-java/ape-client/src/main/java/com/intel/oap/ape/service/netty/client/ParquetDataRequestClient.java

```diff
@@ -90,10 +103,31 @@ public void sendReadBatchRequest(int batchCount) throws IOException, Interrupted
             waitForReaderInitResult();
         }
 
+        requestedBatchCount += batchCount;
+
         tcpChannel.writeAndFlush(new NettyMessage.ReadBatchRequest(batchCount))
                 .addListener(throwErrorListener);
     }
 
+    /**
+     * Wait and return next batch response from remote server.
+     * This will send `readBatch` request to remote server if needed.
+     * @return NettyMessage.ReadBatchResponse
+     * Note: {@link NettyMessage.ReadBatchResponse#releaseBuffers()} should be called after
+     * it's no longer used.
+     */
+    public NettyMessage.ReadBatchResponse nextBatch() throws InterruptedException, IOException {
+        if (requestedBatchCount == 0) {
+            sendReadBatchRequest();
+        }
+
+        NettyMessage.ReadBatchResponse response = responseHandler.nextBatch();
+
+        requestedBatchCount--;

```

Comment:
why need this here?

Reply:
In this class, batch requests can be sent in advance by `sendReadBatchRequest`. 
So `nextBatch` does not send request again if there are sent requests. 
It just waits for result.